### ISSUE TITLE
'getEnclosingBlockScopeContainer' does not handle switch statements and default clauses correctly

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#2281]
Currently `getEnclosingBlockScopeContainer` has the following code:

```
switch (current.kind) {
    case SyntaxKind.SourceFile:
    case SyntaxKind.SwitchKeyword:
    case SyntaxKind.CatchClause:
    case SyntaxKind.ModuleDeclaration:
    case SyntaxKind.ForStatement:
    case SyntaxKind.ForInStatement:
    case SyntaxKind.ForOfStatement:
        return current;
    case SyntaxKind.Block:
        // function block is not considered block-scope container
        // see comment in binder.ts: bind(...), case for SyntaxKind.Block
        if (!isFunctionLike(current.parent)) {
            return current;
        }
}
```

However, I believe `SyntaxKind.SwitchKeyword` should be changed to `SyntaxKind.SwitchStatement` and `SyntaxKind.DefaultClause` should be added in as well.

I would fix it myself, but I'm having trouble figuring a a test case under which the change would be noticeable, since I'm not that familiar with the logic.

**Comment by mhegazy** [comment]
I believe this should be handled by your last change. if so feel free to close it.

**Comment by vladima** [comment]
Fixed by 76dcfb660b85836e19f0e804384fc15c804d24ba
